### ISSUE TITLE
Control where text is injected?

### ISSUE THREAD

**Issue by homerhanumat** [homerhanumat/addinplots#1]
Mentioned by @rpruim:  some code got injected into his .gitignore file.

Would it be possible to check the the type of file that is active, and to inject only if it is, say, .R or .rmd?

**Comment by rpruim** [comment]
My suggestion is simply that under the "done" button some text display informing the user where the injection will occur.  That can help new users understand that injection will happen and also avoid errors that occur when the user mistakenly has the incorrect file open when the process begins.

If you try to restrict to certain types of files, someone will eventually be annoyed that they can't inject into some file they are using that you didn't anticipate.  But perhaps a confirmation dialogue could be used in that situation instead of an absolute block.

**Comment by homerhanumat** [comment]
Yes, I could see some javascript here:  if filetype is not R or rmd, rise
an alert.

**Comment by rpruim** [comment]
Or Rnw

**Comment by homerhanumat** [comment]
yep.

**Comment by homerhanumat** [comment]
Randall,

Dany and I will be at that ShinyDev Conference.  We can ask about some of
those features.

Homer

**Comment by rpruim** [comment]
Wish I could join you.

**Comment by homerhanumat** [comment]
Thinking a bit more about this issue.

The code will be injected into whichever document in the Source window is active, even if the user's focus is elsewhere (e.g., if the user has the cursor in the console).  New users can easily get dinged by this.

The best I can think of now is to add some text to the label of the Done button, so that it reads:

> Done (inject code into active source document)

`rstudioapi::getActiveDocumentContext()` can tell me the filename of the active document, provided it has one.  The observer that handles `input$done` can check the filename extension and cause the app to raise an alert box warning the reader that the active document does not appear to be an R script or R Markdown or Rnw.

How does that sound?
